kernel/aster-nix/src/syscall/nanosleep.rs

```diff
@@ -77,18 +84,39 @@ fn do_clock_nanosleep(
     // current process. i.e., the signals that should be ignored will not interrupt sleeping thread.
     let pauser = Pauser::new();
 
-    let res = pauser.pause_until_or_timeout(|| None, &timeout);
+    let current = current!();
+    let timer_manager = {
+        let clock_id = ClockId::try_from(clockid)?;
+        match clock_id {
+            ClockId::CLOCK_BOOTTIME => BootTimeClock::timer_manager(),
+            ClockId::CLOCK_MONOTONIC => MonotonicClock::timer_manager(),
+            ClockId::CLOCK_REALTIME => RealTimeClock::timer_manager(),
+            // FIXME: We should better not expose this prof timer manager.
+            ClockId::CLOCK_PROCESS_CPUTIME_ID => {
+                current.timer_manager().prof_timer().timer_manager()
+            }
+            // FIXME: From the manual,
+            // the CPU clock IDs returned by clock_getcpuclockid(3)
+            // and pthread_getcpuclockid(3) can also be passed in clockid.
+            // But it's not covered here.
```

Comment:
The related type is `DynamicClockInfo` in `clock_gettime.rs`, but it's ok to leave such clockid as a TODO here.